policy/policy-authority.go

```diff
@@ -139,6 +148,53 @@ func (pa PolicyAuthorityImpl) WillingToIssue(id core.AcmeIdentifier) error {
 	return nil
 }
 
+// CheckCAARecords Does what it says on the tin
+func (pa PolicyAuthorityImpl) CheckCAARecords(id core.AcmeIdentifier) error {
+	domain := id.Value
+	domain = strings.ToLower(domain)
+
+	caaSet, dnssec, err := getCaaSet(domain)
+	if err != nil {
+		err = fmt.Errorf("[DNSSEC: %s] %s", dnssec, err)
+		pa.log.AuditErr(err)
+		fmt.Println(err)
+		return CAAError
+	}
+	if caaSet == nil {
+		pa.log.Audit(fmt.Sprintf("[DNSSEC: %s] Didn't find any CAA records for '%s', can issue", dnssec, domain))
+		return nil
+	}
+	if caaSet.CriticalUnknown() {
+		err = fmt.Errorf("[DNSSEC: %s] Unknown CAA properties flagged as critical for '%s', cannot issue", dnssec, domain)
+		pa.log.AuditErr(err)
+		return UnknownCriticalCAAError
+	}
+
+	if len(caaSet.issue) > 0 || len(caaSet.issuewild) > 0 {
+		var correctSet []*CAA
+		if strings.SplitN(domain, ".", 2)[0] == "*" {
```

Comment:
I *think*, according to the RFC, if a wildcard domain is requested, and there is only an `issue` property but no `issuewild` property, we are still supposed to obey the `issue` property. Is that your understanding?